katcp/inspecting_client.py

```diff
@@ -621,12 +623,23 @@ def inspect_requests(self, name=None, timeout=None):
                     'Error reply during sync process for {}: {}'
                     .format(self.bind_address_string, reply))
 
+        # Get recommended timeouts hints for slow requests if the server
+        # provides them
+        timeout_hints_available = (
+            self.katcp_client.protocol_flags.request_timeout_hints)
+        if timeout_hints_available:
+            timeout_hints = yield self._get_request_timeout_hints(
+                name, timeout=maybe_timeout.remaining())
+        else:
+            timeout_hints = {}
 
         requests_old = set(self._requests_index.keys())
         requests_updated = set()
         for msg in informs:
             req_name = msg.arguments[0]
-            req = {'description': msg.arguments[1]}
+            req = {'name': req_name,
+                   'description': msg.arguments[1],
+                   'timeout_hint': timeout_hints.get(req_name)}
```

Comment:
Yes, that is about the shape of it. I'm requiring the dict key to always be present in the KATCPRequest stuff just to simplify the logic.